sqlite.lisp

```diff
@@ -0,0 +1,122 @@
+(in-package #:cl)
+
+(defpackage #:scheduler-sqlite
+  (:use #:cl #:alexandria #:sqlite #:iterate)
+  (:import-from #:scheduler
+                #:sqlite-scheduler)
+  (:import-from #:scheduler-implementation
+                #:scheduler #:in-memory-scheduler 
+                #:start-scheduler #:stop-scheduler
+                #:create-scheduler-task
+                #:read-scheduler-task
+                #:update-scheduler-task
+                #:delete-scheduler-task
+                #:list-scheduler-tasks
+                ;; entry
+                #:scheduler-entry
+                #:cron-entry
+                #:parse-entry
+                #:parse-cron-entry
+                ;; task
+                #:task
+                #:task-time-specs
+                #:task-command
+                #:task-last-execution
+                #:task-next-execution
+                #:task-source-entry
+                #:execute-task)
+  (:export #:sqlite-scheduler
+           ;; task
+           #:sqlite-task))
+
+(in-package #:scheduler-sqlite)
+
+(defvar *db-file-path*
+  "Path to the SQLite DB file.")
+
+(defclass sqlite-scheduler (scheduler)
+  ())
```

Comment:
I'm not entirely sure about this idea, but perhaps the connection shouldn't be closed after each operation? Some extra burden would be ceded to the user, because then they'd have to call a function responsible for opening and closing the database. What do you think?

Reply:
The burden wouldn't be too much. Everything can be run inside a single `with-open-db-connection`.

Advantages are: 
* less code in the scheduler, 
* scheduler won't need to track connection credentials (for Postgres), 
* less connection setup overhead.

Disadvantages: 
* it will perpetually consume a DB connection, which can be a limited resource in cloud hosted offerings,
* if something goes wrong with the connection during a long running app, recovering from it will require yet more code. More code than currently in the scheduler.
* Using the DB backed scheduler will be even more different from in-memory one than is currently the case.

I don't have serious preference but I do feel current way is cleaner, though less efficient when we get to Postgres connections.

Reply:
I mean - for sqlite. Postgres would require separate deliberation of what would be the best.

re recovery, it could be handled by the macro with-open-db-connection - stop the scheduler and signal an error with a restart "re-connect and re-read all tasks", at least that's how it looks to me without putting much thought into it.

Reply:
What is the advantage you see in opening the sqlite DB outside of scheduler?

Reply:
I don't see an advantage in opening the sqlite database outside the scheduler, but I see an advantage of having only a single connection. Also that seems to be the usual way how sqlite database is used. But this remark is more a thought, I'm not sure myself whether it is a good idea in the first place.

Reply:
OK, let's ignore this idea for now - if you consider it cleaner then so be it.